### DISCUSSION TITLE
error train on VAE after last update

### DISCUSSION THREAD

**Discussion by antorio** [d8ahazard/sd_dreambooth_extension#331] [Q&A]
If I put anything in Pretrained VAE Name or Path, now my logs throws: 
"We couldn't connect to 'https://huggingface.co' to load this model, couldn't find it in the cached files and it looks like .._.._workspace_stable-diffusion-webui_models_VAE is not the path to a directory containing a file named diffusion_pytorch_model.bin"

Checked on generated db_config.json, the string of path is result differently with others, like so:
pretrained_vae_name_or_path:".._.._workspace_stable-diffusion-webui_models_VAE"
instance_data_dir:"../../workspace/excl/temp/"
class_data_dir:"../../workspace/excl/regularization/progamergov/woman_v1-5_mse_vae_ddim50_cfg7_n4420" 

the "/" in string path been overwrited with "_"

am I missing something here in my updates few days ago?

**Comment by antorio** [comment]
For anyone who has same problem as above with the huge updates,
After all the headache, this is temporarily solution for me: **I rolled back to commit #dd2d9aa**
Adding custom VAE is now back working fine.

also, on sidenote:

- last few days the training results is worst than before the updates. In some cases subject/object got mixed up. Also tend to overfit more easily. I tested it using the same dataset and configuration from before the update. At some point the training seem got drifted.

- GPU memory used on my RTX 3090 each training,
before the update: around 45%-60% max
after the update: around >80% (I also got OOM twice because of the auto adjust(WIP) button, without it I'm fine)
after rolling back to previous version: around 45%-60% max

The training seems faster though on the updated version. 
But for now I'm back on track with previous version, better results, and more flexible instance.

**Comment by d8ahazard** [comment]
Give the latest version a try, I've changed a bunch. If it's still puking up errors, let me know and submit an issue if you haven't already.

FWIW, the path to the pretrained VAE should be a directory, not a file.

**Reply by antorio** [reply]
I'm sorry man, but it still throws "wrong pretrained VAE path". I've tried many different path string. Yes it's a directory.
It wasn't a problem before with the same path. 
But so right now I'm not using custom VAE.

**Reply by notchrisb** [reply]
@d8ahazard Hey man, It looks like in you run the `sanitize_filename_part()` function from the images.py module in db_config.py on the   pretrained_vae_name_or_path variable (line 126).  Looks like this is causing the string to convert from 

C:\\Users\\me\\Documents\\git\\stable-diffusion-webui\\models\\VAE

to 

C__Users_me_Documents_git_stable-diffusion-webui_models_VAE

**Reply by antorio** [reply]
So I should just change 
     `   pretrained_vae_name_or_path = images.sanitize_filename_part(pretrained_vae_name_or_path, True) `
        to
      `  pretrained_vae_name_or_path = images.sanitize_filename_part(pretrained_vae_name_or_path, False)`
???

**Reply by antorio** [reply]
it seem not using it altogether solve the problem!
`#  pretrained_vae_name_or_path = images.sanitize_filename_part(pretrained_vae_name_or_path, True)`
probably kinda ubuntu vs windows thing in parsing.
working smoothly now.
